menas/ui/service/EntityValidationService.js

```diff
@@ -0,0 +1,94 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+jQuery.sap.require("sap.m.MessageBox");
+
+var EntityValidationService = new function () {
+
+  this.hasValidHdfsPaths = function (oEntity, sEntityType, oRawInput, oPublishInput) {
+    let isRawPathOk = this._isValidHdfsPath(oEntity.hdfsPath, oRawInput);
+    let isPublishPathOk = this._isValidHdfsPath(oEntity.hdfsPublishPath, oPublishInput);
+
+    if (!isRawPathOk && !isPublishPathOk) {
+      sap.m.MessageToast.show("Please choose the Raw and Publish HDFS paths of the " + sEntityType);
+    } else if (!isRawPathOk) {
+      sap.m.MessageToast.show("Please choose the Raw HDFS path of the " + sEntityType);
+    } else if (!isPublishPathOk) {
+      sap.m.MessageToast.show("Please choose the Publish HDFS path of the " + sEntityType);
+    }
+
+    return isRawPathOk && isPublishPathOk;
+  };
+
+  this.hasValidHdfsPath = function (oEntity, sEntityType, oInput) {
+    let isOk = this._isValidHdfsPath(oEntity.hdfsPath, oInput);
+
+    if (!isOk) {
+      sap.m.MessageToast.show("Please choose the HDFS path of the " + sEntityType);
+    }
+
+    return isOk;
+  };
+
+  this.hasValidName = function (oEntity, sEntityType, oInput) {
+    let isOk = true;
+
+    if (!oEntity.name || oEntity.name === "") {
+      oInput.setValueState(sap.ui.core.ValueState.Error);
+      oInput.setValueStateText(sEntityType + " name cannot be empty");
+      isOk = false;
+    } else if (GenericService.hasWhitespace(oEntity.name)) {
+      oInput.setValueState(sap.ui.core.ValueState.Error);
+      oInput.setValueStateText(
+        sEntityType + " name '" + oEntity.name + "' should not have whitespaces. Please remove spaces and retry");
+      isOk = false;
+    } else if (!oEntity.isEdit && !oEntity.nameUnique) {
+      oInput.setValueState(sap.ui.core.ValueState.Error);
+      oInput.setValueStateText(
+        sEntityType + " name '" + oEntity.name + "' already exists. Choose a different name.");
+      isOk = false;
+    }
+
+    return isOk;
+  };
+
+  this.hasValidSchema = function (oEntity, sEntityType, oNameInput, oVersionInput) {
+    let isOk = true;
+
+    if (!oEntity.schemaName || oEntity.schemaName === "") {
+      oNameInput.setValueState(sap.ui.core.ValueState.Error);
+      oNameInput.setValueStateText("Please choose the Schema of the " + sEntityType);
+      isOk = false;
+    }
+    if (oEntity.schemaVersion === undefined || oEntity.schemaVersion === "") {
+      oVersionInput.setValueState(sap.ui.core.ValueState.Error);
+      oVersionInput.setValueStateText("Please choose the version of the Schema for the " + sEntityType);
+      isOk = false;
+    }
+
+    return isOk;
+  };
+
+  this._isValidHdfsPath = function(sPath, oInput) {
```

Comment:
there are places we use `*HDFS*` and then are places where we use `*Hdfs*`. Could we stick with one in general?